chain/index/read_test.go

```diff
@@ -0,0 +1,307 @@
+	ethTxHash := ethtypes.EthHash([32]byte{1})
+	msgCid := randomCid(t, rng)
+
+	// read from empty db -> ErrNotFound
+	c, err := s.GetCidFromHash(ctx, ethTxHash)
+	require.Error(t, err)
+	require.True(t, errors.Is(err, ErrNotFound))
+	require.EqualValues(t, cid.Undef, c)
+
+	// insert and read
+	insertEthTxHash(t, s, ethTxHash, msgCid)
+	c, err = s.GetCidFromHash(ctx, ethTxHash)
+	require.NoError(t, err)
+	require.EqualValues(t, msgCid, c)
+
+	// look up some other hash -> fails
+	c, err = s.GetCidFromHash(ctx, ethtypes.EthHash([32]byte{2}))
+	require.Error(t, err)
+	require.True(t, errors.Is(err, ErrNotFound))
+	require.EqualValues(t, cid.Undef, c)
+}
+
+func TestGetMsgInfo(t *testing.T) {
+	ctx := context.Background()
+	seed := time.Now().UnixNano()
+	t.Logf("seed: %d", seed)
+	rng := pseudo.New(pseudo.NewSource(seed))
+	s, _, _ := setupWithHeadIndexed(t, 10, rng)
+
+	msgCid := randomCid(t, rng)
+
+	// read from empty db -> ErrNotFound
+	mi, err := s.GetMsgInfo(ctx, msgCid)
+	require.Error(t, err)
+	require.True(t, errors.Is(err, ErrNotFound))
+	require.Nil(t, mi)
+
+	msgCidBytes := msgCid.Bytes()
+	tsKeyCid := randomCid(t, rng)
+	// insert and read
+	insertTipsetMessage(t, s, tipsetMessage{
+		tipsetKeyCid: tsKeyCid.Bytes(),
+		height:       uint64(1),
+		reverted:     false,
+		messageCid:   msgCidBytes,
+		messageIndex: 1,
+	})
+	mi, err = s.GetMsgInfo(ctx, msgCid)
+	require.NoError(t, err)
+	require.Equal(t, msgCid, mi.Message)
+	require.Equal(t, tsKeyCid, mi.TipSet)
+	require.Equal(t, abi.ChainEpoch(1), mi.Epoch)
+}
+
+func setupWithHeadIndexed(t *testing.T, headHeight abi.ChainEpoch, rng *pseudo.Rand) (*SqliteIndexer, *types.TipSet, *dummyChainStore) {
+	head := fakeTipSet(t, rng, headHeight, []cid.Cid{})
+	d := newDummyChainStore()
+	d.SetHeaviestTipSet(head)
+
+	s, err := NewSqliteIndexer(":memory:", d, 0, false, 0)
+	require.NoError(t, err)
+	insertHead(t, s, head, headHeight)
+
+	return s, head, d
+}
+
+func insertHead(t *testing.T, s *SqliteIndexer, head *types.TipSet, height abi.ChainEpoch) {
+	headKeyBytes, err := toTipsetKeyCidBytes(head)
+	require.NoError(t, err)
+
+	insertTipsetMessage(t, s, tipsetMessage{
+		tipsetKeyCid: headKeyBytes,
+		height:       uint64(height),
+		reverted:     false,
+		messageCid:   nil,
+		messageIndex: -1,
+	})
+}
+
+func insertEthTxHash(t *testing.T, s *SqliteIndexer, ethTxHash ethtypes.EthHash, messageCid cid.Cid) {
+	msgCidBytes := messageCid.Bytes()
+
+	res, err := s.stmts.insertEthTxHashStmt.Exec(ethTxHash.String(), msgCidBytes)
+	require.NoError(t, err)
+	rowsAffected, err := res.RowsAffected()
+	require.NoError(t, err)
+	require.Equal(t, int64(1), rowsAffected)
+}
+
+type dummyChainStore struct {
+	mu sync.RWMutex
+
+	heightToTipSet    map[abi.ChainEpoch]*types.TipSet
+	messagesForTipset map[*types.TipSet][]types.ChainMsg
+	keyToTipSet       map[types.TipSetKey]*types.TipSet
+
+	heaviestTipSet   *types.TipSet
+	tipSetByCid      func(ctx context.Context, tsKeyCid cid.Cid) (*types.TipSet, error)
```

Comment:
seems to be unused from the outside, probably remove this and just use a map?

Reply:
Done.